src/backend/distributed/utils/maintenanced.c

```diff
@@ -683,6 +691,109 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 			timeout = Min(timeout, (StatStatementsPurgeInterval * 1000));
 		}
 
+		pid_t backgroundTaskQueueWorkerPid = 0;
+		BgwHandleStatus backgroundTaskQueueWorkerStatus =
+			backgroundTasksQueueBgwHandle != NULL ? GetBackgroundWorkerPid(
+				backgroundTasksQueueBgwHandle, &backgroundTaskQueueWorkerPid) :
+			BGWH_STOPPED;
+		if (!RecoveryInProgress() && BackgroundTaskQueueCheckInterval > 0 &&
+			TimestampDifferenceExceeds(lastBackgroundTaskQueueCheck,
+									   GetCurrentTimestamp(),
+									   BackgroundTaskQueueCheckInterval) &&
+			backgroundTaskQueueWorkerStatus == BGWH_STOPPED)
+		{
+			/* clear old background worker for task queue before checking for new tasks */
+			if (backgroundTasksQueueBgwHandle)
+			{
+				TerminateBackgroundWorker(backgroundTasksQueueBgwHandle);
```

Comment:
we only get here if the background worker is already stopped, right? What does this do?

Reply:
No, we get here in the following chain of events
 - monitor selected task for execution
 - monitor starts background worker
 - task is cancelled
 - monitor updates catalog with pid and running state
 
We need to close the transaction that selects a task for execution before starting the background worker, because otherwise the closing of the transaction will complain about a background worker leaking:
```C
/*
 * The background worker needs to be started outside of the transaction, otherwise
 * it will complain about leaking shared memory segments used, among other things,
 * to communicate the output of the backend.
 */
 ```
 
 This allows a cancel to get the lock on `pg_dist_background_task`. It will find the task `runnable` and not `running` meaning, it will not have any pid to signal. Now the monitor wants to update the task with `running` state and its `pid`. However it finds that the task has been concurrently cancelld.
 
Thus we terminate the backend we just spawned. I was trying to look for a cancel, to keep the behaviour the same, but that would be somewhat convoluted, thus I went for the easy cleanup here.

Reply:
Nevermind, was thinking about the wrong `TerminateBackgroundWorker`.

For this one I blindly copied the logic for the metadataSync: https://github.com/citusdata/citus/blob/79ba490b1f4d590a11b26c2732f9bf6bab89a99b/src/backend/distributed/utils/maintenanced.c#L475-L485

Reply:
Call here has been concluded to be redundant, removed.

Also opened a PR to address the source where this came from: https://github.com/citusdata/citus/pull/6307